aspnetcore/web-api/http-repl.md

```diff
@@ -0,0 +1,597 @@
+
+```console
+https://localhost:5001/people~ pref set colors.json White
+```
+
+Only the [allowed colors](https://github.com/aspnet/HttpRepl/blob/01d5c3c3373e98fe566ff5ef8a17c571de880293/src/Microsoft.Repl/ConsoleHandling/AllowedColors.cs) may be used. Subsequent HTTP requests display output with the new coloring.
+
+### Set the default editor
+
+To test API methods requiring an HTTP request body, a default editor must be set. The HTTP REPL tool launches the configured editor for the sole purpose of composing the request body. Run the following command to set your preferred editor as the default editor:
+
+```console
+pref set editor.command.default "<EXECUTABLE>"
+```
+
+In the preceding command, `<EXECUTABLE>` is the full path to the editor's executable file. For example, run the following command to set Visual Studio Code as the default editor:
+
+# [Linux](#tab/linux)
+
+```console
+pref set editor.command.default "/usr/bin/code"
+```
+
+# [macOS](#tab/macos)
+
+```console
+pref set editor.command.default "/Applications/Visual Studio Code.app/Contents/Resources/app/bin/code"
+```
+
+# [Windows](#tab/windows)
+
+```console
+pref set editor.command.default "C:\Program Files\Microsoft VS Code\Code.exe"
+```
+
+---
+
+## Test the web API
+
+To test a web API, issue one of the following [HTTP requests](https://github.com/microsoft/api-guidelines/blob/vNext/Guidelines.md#74-supported-methods):
+
+* [GET](#http-get-requests)
+* [POST](#http-post-requests)
+* [PUT](#http-put-requests)
+* [DELETE](#http-delete-requests)
+* PATCH
+* HEAD
+* OPTIONS
+
+### HTTP GET requests
+
+To issue an HTTP GET request:
+
+1. Run the `get` command on an endpoint that supports it:
+
+    ```console
+    https://localhost:5001/people~ get
+    ```
+    
+    The preceding command displays the following output format:
+    
+    ```console
+    HTTP/1.1 200 OK
+    Content-Type: application/json; charset=utf-8
+    Date: Fri, 21 Jun 2019 03:38:45 GMT
+    Server: Kestrel
+    Transfer-Encoding: chunked
+    
+    [
+      {
+        "id": 1,
+        "name": "Scott Hunter"
+      },
+      {
+        "id": 2,
+        "name": "Scott Hanselman"
+      },
+      {
+        "id": 3,
+        "name": "Scott Guthrie"
+      }
+    ]
+    
+    
+    https://localhost:5001/people~
+    ```
+
+1. Retrieve a specific record by passing a parameter to the `get` command:
+
+    ```console
+    https://localhost:5001/people~ get 2
+    ```
+    
+    The preceding command displays the following output format:
+    
+    ```console
+    HTTP/1.1 200 OK
+    Content-Type: application/json; charset=utf-8
+    Date: Fri, 21 Jun 2019 06:17:57 GMT
+    Server: Kestrel
+    Transfer-Encoding: chunked
+    
+    [
+      {
+        "id": 2,
+        "name": "Scott Hanselman"
+      }
+    ]
+    
+    
+    https://localhost:5001/people~
+    ```
+
+### HTTP POST requests
```

Comment:
We should note that there are two other ways to do the HTTP requests that require bodies (POST, PUT, PATCH) - you can do it inline:
```
https://localhost:5001/people~ post --content "{ 'id': 0, 'name': 'Scott Addie'}"
```
and you can request it be pulled out of an existing file
```
https://localhost:5001/people~ post --file "c:\whatever.json"
```
Further, if you don't want to send a body (which is rare, so that's why it's not the default) you can do
```
https://localhost:5001/people~ post --no-body
```
If none of those three arguments are specified, that's when the tool will pop up the default editor